runtime/src/main/kotlin/kotlin/util/Sort.kt

```diff
@@ -339,7 +363,7 @@ private fun quickSort(
 internal fun <T> sortArrayWith(
         array: Array<out T>, fromIndex: Int = 0, toIndex: Int = array.size, comparator: Comparator<T>) {
     @Suppress("UNCHECKED_CAST")
-    quickSort(array as Array<T>, fromIndex, toIndex - 1, comparator)
+    mergeSort(array as Array<T>, fromIndex, toIndex - 1, comparator)
```

Comment:
Maybe change the implementation of `sortArrayComparable` as well?

Reply:
Fixed